File: src/client/hlc_lib/typescript/src/hlc.test.ts
// hlc.test.ts — @k1s0/hlc-lib の vitest テスト
// Rust テスト（src/client/hlc_lib/rust/src/lib.rs #[cfg(test)] mod tests）と
// 同等の coverage を TypeScript で実装する。
// 各テストケースは Rust 側の対応するテスト関数と 1:1 の対応を持つ。

// vitest からテストユーティリティをインポートする
import { describe, it, expect } from 'vitest';
// テスト対象の HlcTimestamp・HlcClock をインポートする
import { HlcTimestamp, HlcClock } from './hlc.js';

// HlcTimestamp の順序比較テスト（Rust: test_hlc_timestamp_ordering に対応）
describe('HlcTimestamp 順序比較', () => {
  it('wall_ms が大きい方が後のタイムスタンプであるべき', () => {
    // wall_ms=100 と wall_ms=200 を比較する
    const a = new HlcTimestamp(100n, 0, 0);
    const b = new HlcTimestamp(200n, 0, 0);
    // a < b であることを確認する
    expect(HlcTimestamp.compare(a, b)).toBe(-1);
  });

  it('同一 wall_ms では logical が大きい方が後であるべき', () => {
    // wall_ms=100, logical=0 と wall_ms=100, logical=1 を比較する
    const a = new HlcTimestamp(100n, 0, 0);
    const c = new HlcTimestamp(100n, 1, 0);
    // a < c であることを確認する
    expect(HlcTimestamp.compare(a, c)).toBe(-1);
  });

  it('同一 wall_ms/logical では node_id が大きい方が後であるべき', () => {
    // wall_ms=100, logical=0, node_id=0 と node_id=1 を比較する
    const a = new HlcTimestamp(100n, 0, 0);
    const d = new HlcTimestamp(100n, 0, 1);
    // a < d であることを確認する
    expect(HlcTimestamp.compare(a, d)).toBe(-1);
  });

  it('等値タイムスタンプは 0 を返すべき', () => {
    // 同一フィールドの 2 つのタイムスタンプを比較する
    const a = new HlcTimestamp(100n, 5, 3);
    const b = new HlcTimestamp(100n, 5, 3);
    // compare が 0 を返すことを確認する
    expect(HlcTimestamp.compare(a, b)).toBe(0);
  });
});

// format_compact と parse_compact のラウンドトリップテスト（Rust: test_format_and_parse に対応）
describe('formatCompact / parseCompact ラウンドトリップ', () => {
  it('既知の値で format_compact が正しい文字列を生成すべき', () => {
    // Rust テストと同一の既知値を使用する
    const ts = new HlcTimestamp(0x0123456789abcdefn, 0x00ff, 0x1234);
    // formatCompact を呼び出す
    const formatted = ts.formatCompact();
    // 期待値: Rust テストと同一の形式
    expect(formatted).toBe('0123456789abcdef-00ff-1234');
  });

  it('formatCompact → parseCompact でラウンドトリップが成立すべき', () => {
    // 既知の値で HlcTimestamp を生成する
    const ts = new HlcTimestamp(0x0123456789abcdefn, 0x00ff, 0x1234);
    // format_compact で文字列に変換する
    const formatted = ts.formatCompact();
    // parse_compact でパースする
    const parsed = HlcTimestamp.parseCompact(formatted);
    // パース結果が非 null であることを確認する
    expect(parsed).not.toBeNull();
    // 元の値と等しいことを確認する
    expect(HlcTimestamp.compare(parsed!, ts)).toBe(0);
  });

  it('不正な文字列は null を返すべき', () => {
    // ハイフン区切りが 2 つしかない場合は null を返す
    expect(HlcTimestamp.parseCompact('invalid')).toBeNull();
    // 空文字列は null を返す
    expect(HlcTimestamp.parseCompact('')).toBeNull();
    // パート数が 4 以上の場合は null を返す
    expect(HlcTimestamp.parseCompact('aaa-bbb-ccc-ddd')).toBeNull();
    // 不正な hex 文字列の場合は null を返す
    expect(HlcTimestamp.parseCompact('zzzzzzzzzzzzzzzz-0000-0000')).toBeNull();
  });

  it('EPOCH タイムスタンプのラウンドトリップ', () => {
    // EPOCH（全フィールド 0）のフォーマットとパースを確認する
    const formatted = HlcTimestamp.EPOCH.formatCompact();
    // 期待値: 全て 0 の形式
    expect(formatted).toBe('0000000000000000-0000-0000');
    // パース結果が EPOCH と等しいことを確認する
    const parsed = HlcTimestamp.parseCompact(formatted);
    expect(parsed).not.toBeNull();
    expect(HlcTimestamp.compare(parsed!, HlcTimestamp.EPOCH)).toBe(0);
  });
});

// addMs のテスト（Rust: test_add_ms に対応）
describe('addMs', () => {
  it('wall_ms に duration_ms を正しく加算すべき', () => {
    // 基準タイムスタンプを生成する（Rust テストと同一の値）
    const base = new HlcTimestamp(1_000_000n, 5, 1);
    // 1000ms 後の deadline を計算する
    const deadline = base.addMs(1_000n);
    // wall_ms が正しく加算されることを確認する
    expect(deadline.wall_ms).toBe(1_001_000n);
  });

  it('addMs 後の logical は 0 にリセットされるべき', () => {
    // logical=5 のタイムスタンプから deadline を計算する
    const base = new HlcTimestamp(1_000_000n, 5, 1);
    const deadline = base.addMs(1_000n);
    // logical が 0 にリセットされることを確認する
    expect(deadline.logical).toBe(0);
  });

  it('addMs 後の node_id は引き継がれるべき', () => {
    // node_id=1 のタイムスタンプから deadline を計算する
    const base = new HlcTimestamp(1_000_000n, 5, 1);
    const deadline = base.addMs(1_000n);
    // node_id が引き継がれることを確認する
    expect(deadline.node_id).toBe(1);
  });

  it('U64_MAX 付近のオーバーフローを飽和させるべき', () => {
    // U64_MAX 近くの値からさらに加算する
    const ts = new HlcTimestamp(HlcTimestamp.U64_MAX - 1n, 0, 0);
    // 2 を加算して U64_MAX を超えさせる
    const result = ts.addMs(2n);
    // U64_MAX に飽和することを確認する
    expect(result.wall_ms).toBe(HlcTimestamp.U64_MAX);
  });
});

// elapsedMsSince と isExpiredAt のテスト（Rust: test_elapsed_and_expired に対応）
describe('elapsedMsSince / isExpiredAt', () => {
  it('deadline より前では期限切れにならないべき', () => {
    // 現在時刻を表すタイムスタンプを生成する（Rust テストと同一の値）
    const now = new HlcTimestamp(2_000_000n, 0, 0);
    // 1000ms 後の deadline を計算する
    const deadline = now.addMs(1_000n);
    // now では期限切れでないことを確認する
    expect(deadline.isExpiredAt(now)).toBe(false);
  });

  it('deadline ちょうどで期限切れになるべき', () => {
    // deadline ちょうどで is_expired_at が true を返すことを確認する
    const now = new HlcTimestamp(2_000_000n, 0, 0);
    const deadline = now.addMs(1_000n);
    // deadline ちょうどで期限切れになることを確認する
    expect(deadline.isExpiredAt(deadline)).toBe(true);
  });

  it('deadline を超えたら期限切れになるべき', () => {
    // deadline を超えた future タイムスタンプを生成する
    const now = new HlcTimestamp(2_000_000n, 0, 0);
    const deadline = now.addMs(1_000n);
    const future = new HlcTimestamp(2_001_001n, 0, 0);
    // future では期限切れになることを確認する
    expect(deadline.isExpiredAt(future)).toBe(true);
  });

  it('elapsedMsSince が正しい経過時間を返すべき', () => {
    // Rust テストと同一の値で確認する
    const now = new HlcTimestamp(2_000_000n, 0, 0);
    const future = new HlcTimestamp(2_001_001n, 0, 0);
    // elapsed_ms_since が 1001 を返すことを確認する
    expect(future.elapsedMsSince(now)).toBe(1_001n);
  });

  it('self が reference より前の場合は 0n を返すべき', () => {
    // 逆順（self < reference）の場合の飽和減算を確認する
    const earlier = new HlcTimestamp(1_000n, 0, 0);
    const later = new HlcTimestamp(2_000n, 0, 0);
    // earlier.elapsedMsSince(later) は 0n を返すべき（負の elapsed は表現しない）
    expect(earlier.elapsedMsSince(later)).toBe(0n);
  });
});

// HlcClock.tick の単調増加テスト（Rust: test_tick_monotonic に対応）
describe('HlcClock tick 単調増加', () => {
  it('連続 tick は単調増加タイムスタンプを生成すべき', () => {
    // node_id=0 で HlcClock を生成する
    const clock = new HlcClock(0);
    // 最初の tick を実行する
    let prev = clock.tick();
    // 連続で 99 回 tick して全て単調増加することを確認する（Rust と同一の 100 回）
    for (let i = 0; i < 99; i++) {
      // 次のタイムスタンプを生成する
      const next = clock.tick();
      // prev よりも next が後（または同時）であることを確認する
      expect(HlcTimestamp.compare(next, prev)).toBeGreaterThanOrEqual(0);
      // prev を更新する
      prev = next;
    }
  });

  it('now() は tick() と同等であるべき', () => {
    // node_id=1 で HlcClock を生成する
    const clock = new HlcClock(1);
    // now() の結果が tick() と同じ型（HlcTimestamp）であることを確認する
    const ts = clock.now();
    // HlcTimestamp インスタンスであることを確認する
    expect(ts).toBeInstanceOf(HlcTimestamp);
    // node_id が引き継がれることを確認する
    expect(ts.node_id).toBe(1);
  });
});

// HlcClock.recv の因果関係テスト（Rust: test_recv_causality に対応）
describe('HlcClock recv 因果関係', () => {
  it('recv 後のタイムスタンプは send より後であるべき', () => {
    // 送信者クロックを生成する
    const sender = new HlcClock(1);
    // 受信者クロックを生成する
    const receiver = new HlcClock(2);
    // 送信者で tick する
    const sendTs = sender.tick();
    // 受信者で recv する（send_ts の後になるはずである）
    const recvTs = receiver.recv(sendTs);
    // recv の結果が sendTs 以後であることを確認する
    expect(HlcTimestamp.compare(recvTs, sendTs)).toBeGreaterThanOrEqual(0);
  });

  it('recv は受信メッセージのタイムスタンプから因果関係を引き継ぐべき', () => {
    // 受信者のローカルクロックを初期化する（wall_ms=0）
    const receiver = new HlcClock(0);
    // 未来のタイムスタンプを受信する
    const futureSendTs = new HlcTimestamp(9_999_999n, 100, 1);
    // recv 後のタイムスタンプが futureSendTs 以後であることを確認する
    const recvTs = receiver.recv(futureSendTs);
    // wall_ms が受信メッセージを反映していることを確認する
    expect(recvTs.wall_ms).toBeGreaterThanOrEqual(futureSendTs.wall_ms);
  });
});

// EPOCH 定数テスト（Rust: test_epoch_is_minimum に対応）
describe('EPOCH 定数', () => {
  it('EPOCH は全タイムスタンプの最小値であるべき', () => {
    // 任意の非 EPOCH タイムスタンプを生成する
    const ts = new HlcTimestamp(1n, 0, 0);
    // EPOCH が ts より前であることを確認する
    expect(HlcTimestamp.compare(HlcTimestamp.EPOCH, ts)).toBe(-1);
  });

  it('EPOCH の全フィールドは 0 であるべき', () => {
    // EPOCH の wall_ms が 0n であることを確認する
    expect(HlcTimestamp.EPOCH.wall_ms).toBe(0n);
    // EPOCH の logical が 0 であることを確認する
    expect(HlcTimestamp.EPOCH.logical).toBe(0);
    // EPOCH の node_id が 0 であることを確認する
    expect(HlcTimestamp.EPOCH.node_id).toBe(0);
  });
});

// fromEnv のテスト（Rust: from_env に対応）
describe('HlcClock.fromEnv', () => {
  it('HLC_NODE_ID が未設定の場合は node_id=0 の HlcClock を生成すべき', () => {
    // import.meta.env が存在しない環境では node_id=0 を使用する
    const clock = HlcClock.fromEnv();
    // HlcClock インスタンスであることを確認する
    expect(clock).toBeInstanceOf(HlcClock);
    // tick が動作することを確認する（node_id チェックは内部フィールドにアクセスできないが動作確認）
    const ts = clock.tick();
    expect(ts).toBeInstanceOf(HlcTimestamp);
  });
});

File: src/client/hlc_lib/go/hlc_test.go
// hlc_test.go — hlc パッケージの Go テスト
// Rust テスト（src/client/hlc_lib/rust/src/lib.rs #[cfg(test)] mod tests）と
// 同等の coverage を Go で実装する。
// 各テスト関数は Rust 側の対応するテスト関数と 1:1 の対応を持つ。
package hlc

// testing: Go 標準テストフレームワークをインポートする
import (
	// fmt: エラーメッセージのフォーマットに使用する
	"fmt"
	// sync: concurrent tick テストの WaitGroup に使用する
	"sync"
	// testing: Go テストフレームワーク
	"testing"
)

// TestHlcTimestampOrdering は HlcTimestamp の順序比較を確認するテスト（Rust: test_hlc_timestamp_ordering に対応）
func TestHlcTimestampOrdering(t *testing.T) {
	// wall_ms が大きければ後のタイムスタンプであることを確認する
	a := HlcTimestamp{WallMs: 100, Logical: 0, NodeId: 0}
	b := HlcTimestamp{WallMs: 200, Logical: 0, NodeId: 0}
	// a < b であることを確認する
	if a.Compare(b) >= 0 {
		t.Error("WallMs が大きい方が後であるべき: a.Compare(b) は負であるべき")
	}
	// wall_ms が同じなら logical で比較することを確認する
	c := HlcTimestamp{WallMs: 100, Logical: 1, NodeId: 0}
	if a.Compare(c) >= 0 {
		t.Error("同一 WallMs では Logical が大きい方が後であるべき: a.Compare(c) は負であるべき")
	}
	// logical も同じなら node_id で比較することを確認する
	d := HlcTimestamp{WallMs: 100, Logical: 0, NodeId: 1}
	if a.Compare(d) >= 0 {
		t.Error("同一 WallMs/Logical では NodeId が大きい方が後であるべき: a.Compare(d) は負であるべき")
	}
}

// TestFormatAndParse は FormatCompact と ParseCompact のラウンドトリップを確認するテスト（Rust: test_format_and_parse に対応）
func TestFormatAndParse(t *testing.T) {
	// 既知の値で HlcTimestamp を生成する（Rust テストと同一の値）
	ts := HlcTimestamp{WallMs: 0x0123456789abcdef, Logical: 0x00ff, NodeId: 0x1234}
	// FormatCompact で文字列に変換する
	formatted := ts.FormatCompact()
	// 期待値: Rust テストと同一の形式
	expected := "0123456789abcdef-00ff-1234"
	if formatted != expected {
		t.Errorf("FormatCompact の結果が期待値と一致しない: got=%q, want=%q", formatted, expected)
	}
	// ParseCompact でラウンドトリップが成立することを確認する
	parsed, ok := ParseCompact(formatted)
	if !ok {
		t.Fatalf("ParseCompact が失敗した: input=%q", formatted)
	}
	// 元の値と等しいことを確認する
	if parsed.Compare(ts) != 0 {
		t.Errorf("ラウンドトリップが失敗した: original=%v, parsed=%v", ts, parsed)
	}
}

// TestParseCompactInvalid は ParseCompact が不正な文字列に対して false を返すことを確認するテスト
func TestParseCompactInvalid(t *testing.T) {
	// 不正な形式の文字列のリスト
	invalidInputs := []string{
		// 空文字列
		"",
		// ハイフン区切りが足りない
		"invalid",
		// パート数が足りない
		"0000-0000",
		// 短すぎる wall_ms
		"000-0000-0000",
	}
	// 各不正入力に対して ParseCompact が false を返すことを確認する
	for _, input := range invalidInputs {
		_, ok := ParseCompact(input)
		if ok {
			t.Errorf("不正な入力 %q に対して ParseCompact は false を返すべき", input)
		}
	}
}

// TestAddMs は AddMs が正しく deadline を計算することを確認するテスト（Rust: test_add_ms に対応）
func TestAddMs(t *testing.T) {
	// 基準タイムスタンプを生成する（Rust テストと同一の値）
	base := HlcTimestamp{WallMs: 1_000_000, Logical: 5, NodeId: 1}
	// 1000ms 後の deadline を計算する
	deadline := base.AddMs(1_000)
	// WallMs が正しく加算されることを確認する
	if deadline.WallMs != 1_001_000 {
		t.Errorf("AddMs が WallMs を正しく加算すべき: got=%d, want=%d", deadline.WallMs, uint64(1_001_000))
	}
	// Logical は 0 にリセットされることを確認する
	if deadline.Logical != 0 {
		t.Errorf("AddMs 後の Logical は 0 であるべき: got=%d", deadline.Logical)
	}
	// NodeId は引き継がれることを確認する
	if deadline.NodeId != 1 {
		t.Errorf("AddMs 後の NodeId は引き継ぐべき: got=%d, want=%d", deadline.NodeId, uint16(1))
	}
}

// TestAddMsOverflow は AddMs が MaxUint64 付近でオーバーフローを飽和させることを確認するテスト
func TestAddMsOverflow(t *testing.T) {
	// MaxUint64 - 1 の WallMs を持つタイムスタンプを生成する
	ts := HlcTimestamp{WallMs: ^uint64(0) - 1, Logical: 0, NodeId: 0}
	// 2 を加算して MaxUint64 を超えさせる
	result := ts.AddMs(2)
	// MaxUint64 に飽和することを確認する
	if result.WallMs != ^uint64(0) {
		t.Errorf("AddMs はオーバーフロー時に MaxUint64 に飽和すべき: got=%d", result.WallMs)
	}
}

// TestElapsedAndExpired は ElapsedMsSince と IsExpiredAt の動作を確認するテスト（Rust: test_elapsed_and_expired に対応）
func TestElapsedAndExpired(t *testing.T) {
	// 現在時刻を表すタイムスタンプを生成する（Rust テストと同一の値）
	now := HlcTimestamp{WallMs: 2_000_000, Logical: 0, NodeId: 0}
	// 1000ms 後の deadline を計算する
	deadline := now.AddMs(1_000)
	// now では期限切れでないことを確認する
	if deadline.IsExpiredAt(now) {
		t.Error("deadline より前では期限切れにならないべき")
	}
	// deadline ちょうどで期限切れになることを確認する
	if !deadline.IsExpiredAt(deadline) {
		t.Error("deadline ちょうどで期限切れになるべき")
	}
	// deadline を過ぎたら期限切れになることを確認する
	future := HlcTimestamp{WallMs: 2_001_001, Logical: 0, NodeId: 0}
	if !deadline.IsExpiredAt(future) {
		t.Error("deadline を超えたら期限切れになるべき")
	}
	// ElapsedMsSince の値が正しいことを確認する
	elapsed := future.ElapsedMsSince(now)
	if elapsed != 1_001 {
		t.Errorf("ElapsedMsSince が正しくあるべき: got=%d, want=%d", elapsed, uint64(1_001))
	}
}

// TestElapsedMsSinceSaturation は ElapsedMsSince が self < reference のとき 0 を返すことを確認するテスト
func TestElapsedMsSinceSaturation(t *testing.T) {
	// earlier が later より前のタイムスタンプ
	earlier := HlcTimestamp{WallMs: 1_000, Logical: 0, NodeId: 0}
	later := HlcTimestamp{WallMs: 2_000, Logical: 0, NodeId: 0}
	// earlier.ElapsedMsSince(later) は 0 を返すべき（負の elapsed は表現しない）
	elapsed := earlier.ElapsedMsSince(later)
	if elapsed != 0 {
		t.Errorf("ElapsedMsSince(later) は 0 を返すべき: got=%d", elapsed)
	}
}

// TestTickMonotonic は HlcClock.Tick が単調増加タイムスタンプを生成することを確認するテスト（Rust: test_tick_monotonic に対応）
func TestTickMonotonic(t *testing.T) {
	// nodeId=0 で HlcClock を生成する
	clock := NewHlcClock(0)
	// 連続で 100 回 Tick して全て単調増加することを確認する（Rust と同一の 100 回）
	prev := clock.Tick()
	for i := 0; i < 99; i++ {
		// 次のタイムスタンプを生成する
		next := clock.Tick()
		// prev よりも next が後（または同時）であることを確認する
		if next.Compare(prev) < 0 {
			t.Errorf("Tick は単調増加を保証すべき: prev=%v, next=%v", prev, next)
		}
		// prev を更新する
		prev = next
	}
}

// TestRecvCausality は HlcClock.Recv がメッセージの因果関係を正しく反映することを確認するテスト（Rust: test_recv_causality に対応）
func TestRecvCausality(t *testing.T) {
	// 送信者クロックを生成する
	sender := NewHlcClock(1)
	// 受信者クロックを生成する
	receiver := NewHlcClock(2)
	// 送信者で Tick する
	sendTs := sender.Tick()
	// 受信者で Recv する（sendTs の後になるはずである）
	recvTs := receiver.Recv(sendTs)
	// Recv の結果が sendTs 以後であることを確認する
	if recvTs.Compare(sendTs) < 0 {
		t.Errorf("Recv 後のタイムスタンプは send より後であるべき: send=%v, recv=%v", sendTs, recvTs)
	}
}

// TestConcurrentTick は複数 goroutine で concurrent Tick が単調増加を維持することを確認するテスト（Rust: test_concurrent_tick に対応）
func TestConcurrentTick(t *testing.T) {
	// Arc で HlcClock を共有する（Go では *HlcClock ポインタを直接共有できる）
	clock := NewHlcClock(0)
	// 結果を格納するスライスと Mutex を初期化する
	var mu sync.Mutex
	// 全 goroutine の結果を蓄積するスライス
	all := make([]HlcTimestamp, 0, 1000)
	// WaitGroup で全 goroutine の完了を待つ
	var wg sync.WaitGroup
	// 10 goroutine で並行 Tick を実行する（Rust: 10 スレッドと同等）
	for i := 0; i < 10; i++ {
		wg.Add(1)
		go func() {
			// goroutine 終了時に WaitGroup を通知する
			defer wg.Done()
			// 各 goroutine で 100 回 Tick する（合計 1000 回）
			results := make([]HlcTimestamp, 100)
			for j := 0; j < 100; j++ {
				results[j] = clock.Tick()
			}
			// 結果を all に追加する（Mutex で排他制御）
			mu.Lock()
			all = append(all, results...)
			mu.Unlock()
		}()
	}
	// 全 goroutine の完了を待つ
	wg.Wait()
	// 全タイムスタンプを集合に入れて重複を確認する
	seen := make(map[string]bool, len(all))
	for _, ts := range all {
		// FormatCompact をキーとして重複チェックする
		key := fmt.Sprintf("%016x-%04x-%04x", ts.WallMs, ts.Logical, ts.NodeId)
		if seen[key] {
			t.Errorf("concurrent Tick で重複タイムスタンプが生成された: %s", key)
		}
		seen[key] = true
	}
	// 1000 個（10 goroutine × 100 Tick）のユニークタイムスタンプが生成されたことを確認する
	if len(seen) != 1000 {
		t.Errorf("concurrent Tick は 1000 個の unique タイムスタンプを生成すべき: got=%d", len(seen))
	}
}

// TestEpochIsMinimum は Epoch 定数が最小タイムスタンプであることを確認するテスト（Rust: test_epoch_is_minimum に対応）
func TestEpochIsMinimum(t *testing.T) {
	// 任意の非 Epoch タイムスタンプを生成する
	ts := HlcTimestamp{WallMs: 1, Logical: 0, NodeId: 0}
	// Epoch が ts より前であることを確認する
	if Epoch.Compare(ts) >= 0 {
		t.Error("Epoch は全タイムスタンプの最小値であるべき")
	}
}

// TestNowIsTickAlias は HlcClock.Now が Tick の alias であることを確認するテスト
func TestNowIsTickAlias(t *testing.T) {
	// nodeId=1 で HlcClock を生成する
	clock := NewHlcClock(1)
	// Now を呼び出す
	ts := clock.Now()
	// NodeId が引き継がれることを確認する
	if ts.NodeId != 1 {
		t.Errorf("Now() の NodeId は 1 であるべき: got=%d", ts.NodeId)
	}
}

// TestNewHlcClockFromEnv は NewHlcClockFromEnv が HLC_NODE_ID 未設定時に nodeId=0 の HlcClock を生成することを確認するテスト
func TestNewHlcClockFromEnv(t *testing.T) {
	// HLC_NODE_ID が未設定の環境でテストする（テスト環境では未設定を想定）
	clock := NewHlcClockFromEnv()
	// HlcClock が生成されることを確認する
	if clock == nil {
		t.Fatal("NewHlcClockFromEnv は nil を返すべきでない")
	}
	// Tick が動作することを確認する
	ts := clock.Tick()
	// WallMs が 0 より大きいことを確認する（現在時刻が設定されている）
	if ts.WallMs == 0 {
		t.Error("Tick 後の WallMs は 0 より大きいはず（物理時刻が反映される）")
	}
}
